### ISSUE TITLE
only resolve named placeholders if values is an object

### ISSUE THREAD

**Issue by danieljuhl** [sidorares/node-mysql2#217]
Would it be an idea, to only resolve named placeholders for `options.sql` when type of `options.values` is an object.

The problem is, that I can not set `namedPlaceholder: true` in the general config, and then execute a query like: `conn.execute('SELECT * FROM items WHERE id = ?;', [ id ]);`

I think the `namedPlaceholders` config should simply be to allow for named placeholders, but still work with the other query methods.

If I create tests and PR, are you interested in accepting it and make a release? Or where are you in your release plans.

**Comment by sidorares** [comment]
yes, probably.

you can actually change config after connection ( those flags that don't affect handshake ) but I see your point. @dougwilson - can I ask your thoughts on this?

**Comment by danieljuhl** [comment]
Yes, you are right - and that is my current workaround...

But I don't like, that I keep changing the connections configuration each time I make a query/execute. As I use a pool, I have to explicit set namedPlaceholders on the connection for each request, to make sure it is correct according to my current query.

So hope that we can agree on a better pattern to handle namedPlaceholders.

**Comment by dougwilson** [comment]
Makes sense, I suppose, though adds complexity. Could probably make the rule be array values vs non-array values.

**Comment by danieljuhl** [comment]
My idea would simply be to check if it is an array, otherwise resolve as named params- it does not sound like much complexity imo.

**Comment by gajus** [comment]
I think it is worth considering removing the `namedPlaceholder` configuration in the first place, and use logic: if values is array, use anonymous placeholders, else use named placeholders. This would be a simple change here, https://github.com/sidorares/node-mysql2/blob/936997b69fc432fe0215e328ce42ba29541c100d/lib/connection.js#L455
